src/main/java/org/openrewrite/java/spring/boot3/MigrateWebMvcTagsToObservationConvention.java

```diff
@@ -0,0 +1,285 @@
+                        .build()
+                        .apply(getCursor(), classDecl.getCoordinates().replace());
+
+                J.ClassDeclaration finalNewClassDeclaration = newClassDeclaration;
+                newClassDeclaration = newClassDeclaration
+                        .withLeadingAnnotations(classDecl.getLeadingAnnotations())
+                        .withModifiers(classDecl.getModifiers())
+                        .withPrefix(classDecl.getPrefix())
+                        .withBody(newClassDeclaration.getBody().withStatements(ListUtils.map(classDecl.getBody().getStatements(), stmt -> {
+                            if (stmt instanceof J.MethodDeclaration && ((J.MethodDeclaration) stmt).getSimpleName().equals("getTags")) {
+                                J.MethodDeclaration md = (J.MethodDeclaration) finalNewClassDeclaration.getBody().getStatements().get(0);
+                                md = md.withPrefix(stmt.getPrefix());
+                                //noinspection DataFlowIssue
+                                return md.withBody(md.getBody().withStatements(ListUtils.insertAll(md.getBody().getStatements(), md.getBody().getStatements().size() - 1, getTagsBodyStatements)));
+                            }
+                            return stmt;
+                        })));
+                maybeAddImport(DEFAULTSERVERREQUESTOBSERVATIONCONVENTION_FQ);
+                maybeAddImport(KEYVALUE_FQ);
+                maybeAddImport(KEYVALUES_FQ);
+                maybeAddImport(SERVERREQUESTOBSERVATIONCONVENTION_FQ);
+                maybeRemoveImport(HTTPSERVLETREQUEST_FQ);
+                maybeRemoveImport(HTTPSERVLETRESPONSE_FQ);
+                maybeRemoveImport(TAG_FQ);
+                maybeRemoveImport(TAGS_FQ);
+                maybeRemoveImport(WEBMVCTAGS_FQ);
+                maybeRemoveImport(WEBMVCTAGSPROVIDER_FQ);
+                return (J.ClassDeclaration) super.visitClassDeclaration(newClassDeclaration, ctx);
+            }
+
+            @Override
+            public J.MethodDeclaration visitMethodDeclaration(J.MethodDeclaration method, ExecutionContext ctx) {
+                J.MethodDeclaration m = (J.MethodDeclaration) super.visitMethodDeclaration(method, ctx);
+                J.VariableDeclarations methodParam = (J.VariableDeclarations) m.getParameters().get(0);
+                J.Identifier methodParamIdentifier = methodParam.getVariables().get(0).getName();
+                Boolean addHttpServletResponse = getCursor().pollMessage("addHttpServletResponse");
+                Boolean addHttpServletRequest = getCursor().pollMessage("addHttpServletRequest");
+                if (Boolean.TRUE.equals(addHttpServletResponse) && m.getBody() != null) {
+                    m = JavaTemplate.builder("HttpServletResponse response = #{any()}.getResponse();")
+                            .imports(HTTPSERVLETRESPONSE_FQ, SERVERREQUESTOBSERVATIONCONVENTION_FQ)
+                            .contextSensitive()
```

Comment:
Does this template need to be context sensitive. It doesn't look like it, but this is really hard to tell...